### PR TITLE
[FLINK-34993] parser changes to support model CRUD sql

### PR DESCRIPTION
**Pull request by lihaosky**
## What is the purpose of the change

Support model CRUD sql syntax

## Brief change log

Add following syntax for model
* `create model`
* `drop model`
* `show model`
* `show create model`
* `alter model`

## Verifying this change

Unit test for parser

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (n)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (JavaDocs)

### CODE REVIEW

**Comment by lihaosky** [comment]
cc @twalthr , @wuchong

**Review by twalthr — CHANGES_REQUESTED** [review]
Thanks for working on this @lihaosky. I added some comments. Esp. the support for temporary and error behavior for missing implementation should be solved before merging this.

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/calcite/FlinkPlannerImpl.scala
Comment (lihaosky) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/calcite/FlinkPlannerImpl.scala:151]: It throws like this:
```
org.apache.flink.table.api.TableException: Unsupported query: 
DROP MODEL IF EXISTS M1
```

File: tools/maven/suppressions-parser.xml
Comment (lihaosky) [reply tools/maven/suppressions-parser.xml:24]: I saw some other components have dedicated file. Let me use `suppressions.xml` then

**Comment by lihaosky** [comment]
@twalthr , does this repo have CI build? I can't see it

**Review by twalthr — APPROVED** [review]
I slightly adjusted some names for consistency in the code base. Otherwise this LGTM. I will merge this once the build is green.

**Pull request merged**
